GithubApiWithVIPER/GithubApiWithVIPERTests/GithubApiWithVIPERTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  GithubApiWithVIPERTests.swift
+//  GithubApiWithVIPERTests
+//
+//  Created by Yusuf Özgül on 19.09.2020.
+//  Copyright © 2020 Yusuf Özgül. All rights reserved.
+//
+
+import XCTest
+@testable import GithubApiWithVIPER
+
+/// test cases for repo search module
+class GithubApiWithVIPERTSearchModuleests: XCTestCase {
+    private var presenter: GithubSearchPresenter!
+    private var interactor: GithubSearchInteractor!
+    private var view: MockView!
+    private var router: MockRouter!
+    
+    override func setUp() {
+        interactor = GithubSearchInteractor()
+        view = MockView()
+        router = MockRouter()
+        presenter = .init(view: view, interactor: interactor, router: router)
+        view.presenter = presenter
+    }
+    
+    func testSetPrepareUI() {
+        view.viewdidLoad()
+        XCTAssertEqual(view.prepareUICounts, 1)
+    }
+    
+    func testErroredSearch() {
+        presenter.handleSearchResult(with: .failure(.urlEncode))
+        if view.errors.count == 1 {
+            XCTAssertEqual(view.errors[0], GithubApiError.urlEncode.localizedDescription)
+        } else {
+            XCTFail("View: Unexpected error count")
+        }
+    }
+    
+    func testData() throws {
```

Comment:
test_handleSearchResult_Success_ShouldCheckDataSource method ismi dediğim gibi daha açık olabilir. Onun dışında datayı test edişin gayet güzel 👏

Reply:
searchRepoResult bunu private(set) yaparsan belki bunu da test edebilirsin